blocks/flag.ts

```diff
@@ -75,12 +76,33 @@ const flagBlock: Block<BlockModule<FlagFunc>> = {
       },
     };
     if (isMultivariate(flag)) {
-      const value = ((flag?.variants ?? []).find((variant) =>
-        typeof variant?.rule === "function" && variant?.rule(ctx)
-      ) as Variant<unknown>)?.value ??
-        ((flag?.variants ?? [])[flag?.variants?.length - 1] as Variant<unknown>)
-          ?.value;
-      return isDeferred(value) ? value() : value;
+      let match: Variant<unknown> | null = null;
+
+      for (const variant of flag.variants || []) {
+        if (typeof variant?.rule !== "function") {
+          continue;
+        }
+        let ruleResult = variant?.rule(ctx);
+        // the rule can sometimes be a promise and we need to await it to check if it's truthy or not
+        if (isAwaitable(ruleResult)) {
+          try {
+            ruleResult = await ruleResult;
```

Comment:
in the case you provided (when site.ts has an app middleware) this code runs on serial (one await for each matcher) instead of parallel since all matchers will returns a promise, right? Given that, you should instead, parallelize the rule evaluation.

Reply:
not at all, matchers coming from "website" app continue as `sync` functions, but my local ones, that indirectly depends on my async middleware, turns into `async` function

Reply:
I mean, anyways introducing a middleware will cause all your matchers to be async, so it worth to make parallel instead, wdyt?

Reply:
make sense, your proposal is to always use `Promise.all` on this list, right?

```js
Promise.all(flags.variants.map((variant) => variant.rule(ctx)).find((result) => result)
```